src/hh-apidoc-extensions/PageSections/Guides.php

```diff
@@ -0,0 +1,40 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+namespace HHVM\UserDocumentation\HHAPIDocExt\PageSections;
+
+use type Facebook\HHAPIDoc\PageSections\PageSection;
+use type HHVM\UserDocumentation\GuidesNavData;
+use namespace HH\Lib\{C, Str};
+
+final class Guides extends PageSection {
+
+  <<__Override>>
+  public function getMarkdown(): ?string {
+    $guides = $this->docBlock?->getTagsByName('@guide');
+    if ($guides is null || C\is_empty($guides)) {
+      return null;
+    }
+
+    $markdown = '## Guide';
+    if (C\count($guides) > 1) {
+      $markdown .= 's';
+    }
+    $markdown .= "\n\n";
+
+    foreach ($guides as $path) {
```

Comment:
Can you add tests for this, covering:
- known-bad `@guide` tags
- `@guide tags` that need a redirect

Reply:
The current version leaves all the responsibility on whoever wrote the docblock, we literally just use the @guide string as the URL, so there's not much to test (probably the only meaningful test is that the @guide tags are not ignored like before, I'll add that).

The advantage is that we get redirects etc. for free, but the downside is we can't detect invalid guide URLs or other issues (I noticed on some pages like Vector there are multiple @guide links that redirect to the same place).

I wouldn't be opposed to doing something more fancy, but it gets a little tricky because the API documentation is generated fairly early during the build, before the routing map etc.

Reply:
Ah - but the existing "bad internal link" detection should pick those up after the build?

Reply:
s/detection/unit test/